WordPress/gutenberg
# Build Tooling: Update development tools to work with PostCSS (try 2)

## Description
Fixes #27028.
Previous attempt: #27821.

This should allow to use @wordpress/scripts with PostCSS 8 but doesn't actually require it since it doesn't have it as a dependency at all.

See also https://github.com/webpack-contrib/postcss-loader/blob/v4.1.0/CHANGELOG.md

This PR also further improves the watch options for webpack build as tried in #30343 that caused the revert of #27821.

I also upgraded Storybook to the latest version: `6.2.5` (it was rc.8 before).

## How has this been tested?




`npm run dev` and `npm run build` still work properly,

Create Block still build correctly. Testing is a bit cumbersome but it follows what happens on CI (https://github.com/WordPress/gutenberg/blob/master/bin/test-create-block.sh):

1. `npx wp-create-block esnext-example --no-wp-scripts`
2. `cd esnext-example`
3. `../node_modules/.bin/wp-scripts start`
4. Make sure there are no errors and files are built in `build` folder.
5. `../node_modules/.bin/wp-scripts build`
6. Make sure there are no errors and files are built in `build` folder.
7. `npm run dev` for Gutenberg works as expected.

## Checklist:
